pkg/landscaper/controllers/deployitem/controller.go

```diff
@@ -256,8 +259,10 @@ func (con *controller) detectAbortingTimeouts(log logging.Logger, di *lsv1alpha1
 	return &requeue, nil
 }
 
-func (con *controller) detectProgressingTimeouts(log logging.Logger, di *lsv1alpha1.DeployItem) (*time.Duration, error) {
-	logger := log.WithValues("operation", "DetectProgressingTimeouts")
+func (con *controller) detectProgressingTimeouts(ctx context.Context, di *lsv1alpha1.DeployItem) (*time.Duration, error) {
+	logger, _ := logging.FromContextOrNew(ctx, nil)
+	logger = logger.WithValues(lc.KeyMethod, "detectProgressingTimeouts")
```

Comment:
```suggestion
	logger, ctx := logging.FromContextOrNew(ctx, nil, lc.KeyMethod, "detectProgressingTimeouts")
```

Reply:
Again the `KeyMethod` logging key, which is only for the current method, but should not be passed in a context to sub methods.